### PR TITLE
Autosaves per Game Name - Quality of Life Enhancement

### PR DESCRIPTION
**Pull request by mattthewaz**
When I play multiplayer games, I often cycle between files frequently depending on who I'm playing with. I have a file here that I play with one friend, another there that I player with a couple others, so on and so forth. With all this switching around, it can be easy to forget to save a Replay and have all traces of that file overwritten as your last hope - the autosaves - are overwritten by the other games you're hosting.
This change solves that by including the name of the game session in the autosave's name. Each name will support the usual 5 slots as per normal. For example, if I host a game and give it the name "Together", then the autosaves created from this will be named "Autosave-Together-1", "Autosave-Together-2", etc... and will not overwrite other autosaves from sessions with other names.
Note: the autosave's use the name of the game session (the name picked in the host dialog), not the name of the loaded file/replay.

autosavePerGameName

### CODE REVIEW

**Comment by mattthewaz** [comment]
For testing, I did confirm that it still obeys the 5 autosave limit; it will find the oldest autosave with a matching name to overwrite when it runs out of slots.

**Comment by jacobEAdamson** [comment]
I'm torn because autosaves currently work like they do in vanilla. Let me talk with some of the other devs to make sure we want this feature. Thanks for coding it by the way.

**Comment by mattthewaz** [comment]
If nothing else a toggle can be added for this behavior.

**Comment by jacobEAdamson** [comment]
Yeah, if you can gate it behind a setting in the multiplayer mod, that would be good

**Comment by tindolt-zz** [comment]
Add a setting to turn that on and off and I agree it would be kinda nice.

**Review by tindolt-zz — CHANGES_REQUESTED** [review]
Add mod setting toggle for turning this feature on and off and it should be good, have it default to off.

**Comment by mattthewaz** [comment]
I have added the requested toggle to the Host dialog, defaulted it to 'off' to maintain the vanilla user experience, and added an entry to each Language file for the new option.

Manually tested and confirmed that the toggle works as expected.

**Review by tindolt-zz — APPROVED** [review]
Looks great! I am sure several people will like this feature.

**Comment by MilanDierick** [comment]
I resolved a merge conflict in the German translation file.

**Review by SamboyCoding — APPROVED** [review]
Looks good to me!

**Review by MilanDierick — DISMISSED** [review]
Please merge master branch into autosavesPerGameName branch, I just merged tons of code refactoring.

**Comment by mattthewaz** [comment]
@MilanDierick , I have merged master into the autosavesPerGameName, as requested.

**Comment by MilanDierick** [comment]
@mattthewaz Another issue would be that most people just leave the name as default when hosting, which means when a player hosts multiple games with the same name, how do you differentiate between the different auto saves? I propose that, in case the player chose to use the default name, to keep track of the auto saves, and when the player saves the replay, to rename the auto saves so they no longer use the default name, and thus prevent auto save name clashes when playing multiple multiplayer games at the same time.

**Comment by mattthewaz** [comment]
I agree that my solution would not help people who do not change the default name; some level of user action is involved if they want to use the feature effectively. Here are the options as I see them:

- Change nothing for now. Additional rules constitute scope creep and might over-complicate the feature. 
- Follow your suggestion. We would detect when the Host has used the default name and use the Replay name when creating the filenames. In case the game wasn't loaded from Replay, still use the default name in the autosave.
- Simplified version of your suggestion. We could always use the Replay name if one exists and only use the game name when one does not.

**Comment by mattthewaz** [comment]
I personally feel like the enhancement as it stands is a good starting point. It would be easy to iterate on this later.

File: Source/Client/Windows/HostWindow.cs [hunk 1]
```diff
@@ -104,6 +104,13 @@ public override void DoWindowContents(Rect inRect)
 
             entry = entry.Down(30);
 
+            var appendNameToAutosaveLabel = $"{"MpAppendNameToAutosave".Translate()}:  ";
```
Comment (jacobEAdamson) [inline Source/Client/Windows/HostWindow.cs]: So, mostly I like the idea, but I'd really prefer not to clutter the host page with this option. The host game screen should only be for options that make sense on a session by session basis. Is it possible to move this to the mod settings located in the `MultiplayerMod.cs` file?

**Review by jacobEAdamson — CHANGES_REQUESTED** [review]

File: Source/Client/Windows/HostWindow.cs (hunk 1 above)
Comment (mattthewaz) [reply Source/Client/Windows/HostWindow.cs]: @jacobEAdamson I agree, moving this out of the Host dialog and into Mod settings sounds like a good idea.

Comment (mattthewaz) [reply Source/Client/Windows/HostWindow.cs]: I'll work on that when I get home later tonight

Comment (mattthewaz) [reply Source/Client/Windows/HostWindow.cs]: This change has been made.
autosavePerGameNameModSettings

**Comment by mattthewaz** [comment]
Addressed merge conflicts.

**Review by MilanDierick — APPROVED** [review]
Looks good to me.

**Comment by mattthewaz** [comment]
I have targeted the pull request to go to development, as per the process

**Review by jacobEAdamson — APPROVED** [review]

**Pull request merged**